sql/src/main/java/io/crate/operation/udf/UserDefinedFunctionMetaData.java

```diff
@@ -73,6 +74,17 @@ public String name() {
         return arguments;
     }
 
+    public List<DataType> argumentDataTypes() {
+        return arguments.stream().map(FunctionArgumentDefinition::type).collect(toList());
```

Comment:
Think this should be saved in the ctor - it's called in sameSignature which is called inside a loop so you might have the list allocations here multiple times.